features/proxyNew/state/proxyStateMachine.types.ts

```diff
@@ -0,0 +1,76 @@
+import { Observable } from 'rxjs'
+import { spawn } from 'xstate'
+
+import { TxHelpers } from '../../../components/AppContext'
+import { HasGasEstimation } from '../../../helpers/form'
+import { createProxyStateMachine } from './proxyStateMachine'
+
+export interface ProxyContext {
+  readonly txHelper: TxHelpers
+  readonly proxyAddress$: Observable<string | undefined>
+  readonly getGasEstimation$: (estimatedGasCost: number) => Observable<HasGasEstimation>
+  readonly safeConfirmations: number
+  gasData?: HasGasEstimation
+  txHash?: string
+  txError?: string
+  proxyConfirmations?: number
+  proxyAddress?: string
+}
```

Comment:
What do you think of grouping this by dependencies and machine-state?

```suggestion
export interface ProxyContext {
  deps: {
    readonly txHelper: TxHelpers
    readonly proxyAddress$: Observable<string | undefined>
    readonly getGasEstimation$: (estimatedGasCost: number) => Observable<HasGasEstimation>
    readonly safeConfirmations: number
  },
  inMachine: {
    gasData?: HasGasEstimation
    txHash?: string
    txError?: string
    proxyConfirmations?: number
    proxyAddress?: string
  }
}
```

`inMachine` is a pretty crappy name but what do you think of the idea?

Reply:
great idea!

Reply:
I put the dependencies in a separated node, but it will be better to keep the state in root because we can't simply `assign` nested object:
	```javascript
	assign((context, event) => ({  
	  machineState: {  
	    proxyAddress: event.proxyAddress,  
	    totalSteps: event.proxyAddress ? 2 : 3,  
	  },  
	})),
	```
The code will replace `machineState` instead of merging with the initial state